scipy/stats/_distn_infrastructure.py

```diff
@@ -612,15 +611,18 @@ def _parse_args_stats(self, %(shape_arg_str)s %(locscale_in)s, moments='mv'):
 # I think the function name ncx2 is an abbreviation for noncentral chi squared.
 
 def _ncx2_log_pdf(x, df, nc):
-    a = asarray(df/2.0)
-    fac = -nc/2.0 - x/2.0 + (a-1)*log(x) - a*log(2) - gammaln(a)
-    return fac + np.nan_to_num(log(hyp0f1(a, nc * x/4.0)))
-
+    # We use (xs**2 + ns**2)/2 = (xs - ns)**2/2  + xs*ns, and include the factor
+    # of exp(-xs*ns) into the ive function to improve numerical stability
+    # at large values of xs. See also `rice.pdf`.
+    df2 = df/2.0 - 1.0
+    xs, ns = np.sqrt(x), np.sqrt(nc)
+    res = xlogy(df2/2.0, x/nc) - 0.5*(xs - ns)**2
+    res += np.log(ive(df2, xs*ns) / 2.0)
+    return res
 
 def _ncx2_pdf(x, df, nc):
     return np.exp(_ncx2_log_pdf(x, df, nc))
 
-
```

Comment:
Don't remove this blank line.  See the first sentence in the "Blank Lines" section of PEP 8: https://www.python.org/dev/peps/pep-0008/#blank-lines